src/rgw/rgw_sal_motr.cc

```diff
@@ -2967,7 +2970,34 @@ int MotrStore::get_user_by_access_key(const DoutPrefixProvider *dpp, const std::
 
 int MotrStore::get_user_by_email(const DoutPrefixProvider *dpp, const std::string& email, optional_yield y, std::unique_ptr<User>* user)
 {
-  return 0;
+  int rc;
+  User *u;
+  bufferlist bl;
+  RGWUserInfo uinfo;
+  
+  rc = do_idx_op_by_name(RGW_IAM_MOTR_EMAIL_KEY,
+                           M0_IC_GET, email, bl);
+  if (rc < 0){
+    ldout(cctx, 0) << "Email Id not found: rc = " << rc << dendl;
+    return rc;
+  }
+
+  bufferlist& blr = bl;
+  auto iter = blr.cbegin();
+  uinfo.decodeUserByEmail(iter);
```

Comment:
For function names, Camel Casing is not used in ceph I think.
`decode_user_by_email` might be better.

Reply:
Fixed